services/datamanager/datacapture/data_capture_file.go

```diff
@@ -37,6 +39,10 @@ type File struct {
 	writer   *bufio.Writer
 	size     int64
 	metadata *v1.DataCaptureMetadata
+
+	initialReadOffset int64
```

Comment:
These were added because we previously had a bug: as the file was written/read, the (shared) file descriptor pointer would be updated. So if you read data then had to read it again (say because the upload failed and needed to be retried), you'd get an immediate EOF (because the pointer would already be at the end of the file).

Reply:
This caused a bug where exponentialRetries didn't work (because after a single failure, all ensuing calls would also fail). This is now fixed.